meson.build

```diff
@@ -0,0 +1,1207 @@
+])
+
+# PJSIP_SIMPLE
+
+pjsip_simple = library('pjsip-simple', [
+	'pjsip/src/pjsip-simple/errno.c',
+	'pjsip/src/pjsip-simple/evsub.c',
+	'pjsip/src/pjsip-simple/evsub_msg.c',
+	'pjsip/src/pjsip-simple/iscomposing.c',
+	'pjsip/src/pjsip-simple/mwi.c',
+	'pjsip/src/pjsip-simple/pidf.c',
+	'pjsip/src/pjsip-simple/presence.c',
+	'pjsip/src/pjsip-simple/presence_body.c',
+	'pjsip/src/pjsip-simple/publishc.c',
+	'pjsip/src/pjsip-simple/rpid.c',
+	'pjsip/src/pjsip-simple/xpidf.c',
+], link_with: [
+	pjlib,
+	pjlib_util,
+	pjsip,
+], include_directories: [
+	pjlib_inc,
+	pjlib_util_inc,
+	pjsip_inc,
+])
+
+# PJSIP_UA
+
+pjsip_ua = library('pjsip-ua', [
+	'pjsip/src/pjsip-ua/sip_inv.c',
+	'pjsip/src/pjsip-ua/sip_reg.c',
+	'pjsip/src/pjsip-ua/sip_replaces.c',
+	'pjsip/src/pjsip-ua/sip_xfer.c',
+	'pjsip/src/pjsip-ua/sip_100rel.c',
+	'pjsip/src/pjsip-ua/sip_timer.c',
+], link_with: [
+	pjlib,
+	pjlib_util,
+	pjsdp,
+	pjsip,
+	pjsip_simple,
+], include_directories: [
+	pjlib_inc,
+	pjlib_util_inc,
+	pjsip_inc,
+	pjmedia_inc,
+])
+
+# PJSUA_LIB
+
+pjsua_lib = library('pjsua-lib', [
+	'pjsip/src/pjsua-lib/pjsua_acc.c',
+	'pjsip/src/pjsua-lib/pjsua_call.c',
+	'pjsip/src/pjsua-lib/pjsua_core.c',
+	'pjsip/src/pjsua-lib/pjsua_im.c',
+	'pjsip/src/pjsua-lib/pjsua_media.c',
+	'pjsip/src/pjsua-lib/pjsua_pres.c',
+	'pjsip/src/pjsua-lib/pjsua_dump.c',
+	'pjsip/src/pjsua-lib/pjsua_aud.c',
+	'pjsip/src/pjsua-lib/pjsua_vid.c',
+], link_with: [
+	pjlib,
+	pjlib_util,
+	pjmedia,
+	pjsip,
+	pjsip_simple,
+	pjnath,
+	pjmedia_codec,
+	pjmedia_audiodev,
+	pjsip_ua,
+], include_directories: [
+	pjlib_inc,
+	pjlib_util_inc,
+	pjsip_inc,
+	pjmedia_inc,
+	pjnath_inc,
+])
+
+# TEST
+
+test('PJSIP Test', executable('pjsip-test', [
+	'pjsip/src/test/main.c',
+	'pjsip/src/test/dlg_core_test.c',
+	'pjsip/src/test/dns_test.c',
+	'pjsip/src/test/msg_err_test.c',
+	'pjsip/src/test/msg_logger.c',
+	'pjsip/src/test/msg_test.c',
+	'pjsip/src/test/multipart_test.c',
+	'pjsip/src/test/regc_test.c',
+	'pjsip/src/test/test.c',
+	'pjsip/src/test/transport_loop_test.c',
+	'pjsip/src/test/transport_tcp_test.c',
+	'pjsip/src/test/transport_test.c',
+	'pjsip/src/test/transport_udp_test.c',
+	'pjsip/src/test/tsx_basic_test.c',
+	'pjsip/src/test/tsx_bench.c',
+	'pjsip/src/test/tsx_uac_test.c',
+	'pjsip/src/test/tsx_uas_test.c',
+	'pjsip/src/test/txdata_test.c',
+	'pjsip/src/test/uri_test.c',
+	'pjsip/src/test/inv_offer_answer_test.c',
+], dependencies: [
+	m_dep,
+], link_with: [
+	pjlib,
+	pjlib_util,
+	pjsdp,
+	pjsip,
+	pjsip_ua,
+], include_directories: [
+	pjlib_inc,
+	pjlib_util_inc,
+	pjsip_inc,
+	pjmedia_inc,
+]), timeout: 0, workdir: join_paths(meson.current_source_dir(), 'pjsip/build') )
+
+#
+
+pjproject_dep = declare_dependency(link_with : [
+	pjlib,
+	pjlib_util,
+	pjsip,
+	pjsip_ua,
+	pjsip_simple,
+	pjmedia,
+	pjmedia_codec,
+	pjmedia_audiodev,
+	pjnath,
+	pjsua_lib,
+], include_directories : [
+	pjlib_inc,
+	pjlib_util_inc,
+	pjmedia_inc,
+	pjnath_inc,
+	pjsip_inc,
+])
+
+# PJSUA-APPS
+
```

Comment:
Might be worth making these into a for loop?

```meson
apps = [
    'auddemo', 'aviplay', 'aectest', 'clidemo',
    'confsample', # .....
]
foreach app: apps
    executable(app, 'pjsip-apps/src/samples/@0@.c'.format(app), dependencies: pjproject_dep)
endforeach

Reply:
Yep, it may. All of these suggested ideas look like optimizations and improvements.

I generally welcome advanced use of meson tools, but first I would like to understand how generally able idea of using `meson` in `PJPROJECT` as a alternative build system.

If the welcome `meson` integration step is successful pass, we can continue to improve and extend `meson.build` in accordance with best practices and provie more architecture and platofrms.

At this stage, this is the first step (actually second) I would like to show support Linux and macOS (actually by direct PJPROJECT request and does not have any plan about support macOS platform) clarify for myself in general the possibility of using the `pjproject` as an external project in my source code.

Since I does not have feedback about this step I also does not want to spend more steps. Let I remind that the issue of integration with meson was raised about two years ago and while we are still on the beginning of road.